homeassistant/components/thermostat/proliphix.py

```diff
@@ -0,0 +1,206 @@
+"""homeassistant.components.thermostat.proliphix
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Proliphix NT10e Thermostat is an ethernet connected thermostat. It
+has a local HTTP interface that is based on get/set of OID values. A
+complete collection of the API is available in this API doc:
+
+https://github.com/sdague/thermostat.rb/blob/master/docs/PDP_API_R1_11.pdf
+"""
+
+import logging
+import requests
+from urllib.parse import urlencode
+from homeassistant.components.thermostat import (ThermostatDevice, STATE_COOL,
+                                                 STATE_IDLE, STATE_HEAT)
+from homeassistant.const import (CONF_USERNAME, CONF_PASSWORD,
+                                 CONF_HOST, TEMP_FAHRENHEIT)
+
+logger = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the proliphix thermostats. """
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
+
+    add_devices([
+        ProliphixThermostat(host, username, password)
+    ])
+
+
+# The subset of oids which are useful for home-assistant
+OIDS = {
+    '1.2': 'DevName',
+    '4.1.13': 'AverageTemp',
+    '4.1.4': 'FanState',
+    '4.1.5': 'SetbackHeat',
+    '4.1.6': 'SetbackCool',
+    '4.1.2': 'HvacState',
+    '4.1.11': 'CurrentClass',
+    '4.5.1': 'Heat1Usage',
+    '4.5.3': 'Cool1Usage',
+    '4.5.5': 'FanUsage',
+    '4.5.6': 'LastUsageReset'
+}
+
+
+def _get_oid(name):
+    """Get OID by name/value."""
+    for oid, value in OIDS.items():
+        if value == name:
+            return oid
+    return None
+
+
+def _all_oids():
+    """Build a query string for all the OIDS we have."""
+    return "&".join([("OID" + x + "=") for x in OIDS.keys()])
+
+
+class PDP(object):
```

Comment:
Could you extract this code into a standalone library that will be installed when the platform is loaded using the `REQUIREMENTS` variable?